### PR TITLE
i2c changes for STM32F4

### PR DESCRIPTION
**Pull request by blckmn**

### CODE REVIEW

**Comment by rs2k** [comment]
Is this ready to merge in?

**Comment by blckmn** [comment]
Good to go. Tested baro and it worked no issues. Can't test AlienFlight or RevoNano though.

**Comment by MJ666** [comment]
@blckmn the new I2C driver looks quite good and much more clean now. I also was digging into this last days. In my understanding the F1 and the F4 have very similar peripherals. I have seen you removed the i2cUnstick() code. I know from some F1 examples the I2C bus can lookup for some reasons and this function was used during the error handling to unlock the I2C bus. Normal reinitialization was not always sufficient. May be it is more save to keep this function. Our AlienFlightF4 prototypes have no I2C peripherals yet but this will likely come soon for the new variants. We also think about to connect all sensors via SPI.

@rs2k looking forward to see this merged. This will likely also solve a few issues I have seen with the existing driver for the AlienFlightF4 during my early testing with the I2C bus.

**Comment by blckmn** [comment]
I'll add the unstick back in (might tidy it up slightly too) and fix up the merge conflicts

File: src/main/target/ALIENFLIGHTF4/target.h [hunk 1]
```diff
@@ -129,7 +129,14 @@
 
 //#define USE_I2C
 #define I2C_DEVICE (I2CDEV_1)
-//#define I2C_DEVICE_EXT (I2CDEV_2)
```
Comment (MJ666) [inline src/main/target/ALIENFLIGHTF4/target.h:132]: Please keep this line in for now.

**Pull request closed without merging**